base/applications/taskmgr/graphctl.c

```diff
@@ -12,666 +12,363 @@
-
-    /*  if we don't have one yet, set up a memory dc for the grid */
-    if (this->m_dcGrid == NULL)
+    HDC    hdcg;
+    PBYTE  t;
+    RECT   rcDirt;
+
+    UINT   Prev0, Prev1, RetainingWidth;
+    INT    PrevY, CurrY, p, v;
+
+    hdcg = inst->hdcGraph;
+    RetainingWidth = inst->BitmapWidth - PLOT_SHIFT;
+    t = inst->PointBuffer;
+    Prev0 = *(t + inst->CurrIndex);
+    Prev1 = *(t + inst->CurrIndex + inst->NumberOfPoints);
+    if (inst->CurrIndex < inst->NumberOfPoints)
     {
-        this->m_dcGrid = CreateCompatibleDC(dc);
-        this->m_bitmapGrid = CreateCompatibleBitmap(dc, this->m_nClientWidth, this->m_nClientHeight);
-        this->m_bitmapOldGrid = (HBITMAP)SelectObject(this->m_dcGrid, this->m_bitmapGrid);
+        inst->CurrIndex++;
     }
-    else if(bResize)
+    else
     {
-        // the size of the drawing area has changed
-        // so create a new bitmap of the appropriate size
-        if(this->m_bitmapGrid != NULL)
-        {
-            this->m_bitmapGrid = (HBITMAP)SelectObject(this->m_dcGrid, this->m_bitmapOldGrid);
-            DeleteObject(this->m_bitmapGrid);
-            this->m_bitmapGrid = CreateCompatibleBitmap(dc, this->m_nClientWidth, this->m_nClientHeight);
-            SelectObject(this->m_dcGrid, this->m_bitmapGrid);
-        }
+        inst->CurrIndex = 0;
     }
-
-    SetBkColor(this->m_dcGrid, this->m_crBackColor);
-
-    /*  fill the grid background */
-    FillRect(this->m_dcGrid, &this->m_rectClient, this->m_brushBack);
-
-    /*  draw the plot rectangle: */
-    /*  determine how wide the y axis scaling values are */
-    nCharacters = abs((int)log10(fabs(this->m_dUpperLimit)));
-    nCharacters = max(nCharacters, abs((int)log10(fabs(this->m_dLowerLimit))));
-
-    /*  add the units digit, decimal point and a minus sign, and an extra space */
-    /*  as well as the number of decimal places to display */
-    nCharacters = nCharacters + 4 + this->m_nYDecimals;
-
-    /*  adjust the plot rectangle dimensions */
-    /*  assume 6 pixels per character (this may need to be adjusted) */
-    /*   m_rectPlot.left = m_rectClient.left + 6*(nCharacters); */
-    this->m_rectPlot.left = this->m_rectClient.left;
-    this->m_nPlotWidth    = this->m_rectPlot.right - this->m_rectPlot.left;/* m_rectPlot.Width(); */
-
-    /*  draw the plot rectangle */
-    oldPen = (HPEN)SelectObject(this->m_dcGrid, solidPen);
-    MoveToEx(this->m_dcGrid, this->m_rectPlot.left, this->m_rectPlot.top, NULL);
-    LineTo(this->m_dcGrid, this->m_rectPlot.right + 1, this->m_rectPlot.top);
-    LineTo(this->m_dcGrid, this->m_rectPlot.right + 1, this->m_rectPlot.bottom + 1);
-    LineTo(this->m_dcGrid, this->m_rectPlot.left, this->m_rectPlot.bottom + 1);
-    /*   LineTo(m_dcGrid, m_rectPlot.left, m_rectPlot.top); */
-
-    /*  draw the horizontal axis */
-    for (i = this->m_rectPlot.top; i <= this->m_rectPlot.bottom; i += this->m_nGridYPixels)
+    *(t + inst->CurrIndex) = val0;
```

Comment:
What is "val0" / "val1" ? More descriptive names?

Reply:
because this "control" is somewhat generalized, despite being pretty specific to the perfpage, `val` is the best description, because for one instance, it's CpuUsage, for another it's memory related percent. is `nBarsUsed` (in perfpage.c) more descriptive? I doubt it.